config/config.go

```diff
@@ -36,19 +50,52 @@ func expandHome(path string) (string, error) {
 	}
 }
 
-func ReadConfig() Config {
-	var config Config
-	if err := json.Unmarshal(rawConfig, &config); err != nil {
-		zap.L().Fatal("failed to read config.json", zap.Error(err))
+type (
+	Config struct {
+		Peers []*PeerConfig
+		Paths struct {
+			FuseDir   string `json:"fuse_dir"`
+			MirrorDir string `json:"mirror_dir"`
+		} `json:"paths"`
 	}
+	PeerConfig struct {
+		Host string
+		Name string
+		Port uint
+	}
```

Comment:
I wonder whether it's a good idea to leave the struct untagged. Just to be extra sure, I always annotate structs when I intend to serialize/deserialize them but it's more of a habit than something purposeful.

Reply:
fair point, I do that too, idk why I didn't do it here :P good catch